src/client/Microsoft.Identity.Client/Cache/Items/MsalAccessTokenCacheItem.cs

```diff
@@ -30,7 +30,7 @@ internal class MsalAccessTokenCacheItem : MsalCredentialCacheItemBase
             string keyId = null,
             string oboCacheKey = null)
             : this(
-                scopes: response.Scope, // token providers send pre-sorted (alphabetically) scopes
+                scopes: ScopeHelper.OrderScopesAlphabetically(response.Scope), // order scopes to avoid cache duplication. This is not in the hot path.

```

Comment:
Does it solve #4474 by normalizing the token cache key to use an ordered scope string?

For what it's worth, I brought up this topic before, but it was considered too late to change the specs. That being said, I still think the token cache key shall use normalized/ordered scope, which [will give token search a performance boost](https://github.com/AzureAD/microsoft-authentication-library-for-python/blame/1.26.0/msal/token_cache.py#L96-L98). Token cache compatibility will not really be affected by normalizing AT's key.

Reply:
> Does it solve https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4474 by normalizing the token cache key to use an ordered scope string?

Yes. 

The assumption was that AAD always returns sorted scopes, while that seems to usually happen, it's not a strict contact (and order is only for non-OAuth scopes).

Reply:
The scopes in the token response are ordered for Azure related IdP (AAD, B2C, CIAM), but not for other identity providers.

Reply:
> The scopes in the token response are ordered for Azure related IdP (AAD, B2C, CIAM), but not for other identity providers.

In that case, MSAL might as well just normalize (i.e., sort) the scopes before writing the key into token cache. Is that this PR already doing? Other MSALs can follow suit. @bgavrilMS

Reply:
Yes, this is what this fix is doing. But I don't think you can optimize the token lookup logic, because of older token caches.